### PR TITLE
[Provisioner] Support docker in Lambda Cloud and TPU

### PR DESCRIPTION
**Pull request by cblmemo**
Support docker as runtime on Lambda Cloud and TPU. TPU part is inherit from #3819.

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
     - [x] `sky launch --image-id docker:continuumio/miniconda3:4.12.0 --cloud lambda`
     - [x] `sky launch --image-id docker:continuumio/miniconda3:4.12.0 --cloud azure`
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: sky/provision/docker_utils.py
Comment (Michaelvll) [inline sky/provision/docker_utils.py:203]: Can we keep this command running within other commands, e.g., _check_docker_installed, to reduce the ssh connection?

Comment (Michaelvll) [reply sky/provision/docker_utils.py:203]: Also, just curious, how does it compare with using `sudo docker` in all places `docker` is called. Asking because had the same issue before running docker on TPU instances.

**Review by Michaelvll — APPROVED** [review]
Thanks for adding the support @cblmemo! Just want to make sure that `id -nG $USER | grep -qw docker` can be successfully run on other clouds as well.

File: sky/provision/docker_utils.py
Comment (cblmemo) [reply sky/provision/docker_utils.py]: Lets merge this version first to unblock our user ;) filed an issue #4121 to keep track of this.

**Comment by Michaelvll** [comment]
> > Thanks for adding the support @cblmemo! Just want to make sure that `id -nG $USER | grep -qw docker` can be successfully run on other clouds as well. Can we test with AWS CPU/GPU, GCP CPU/GPU/TPU to make sure it does not break existing services?
> 
> Do you mean testing them with docker? As the codepath changed by this PR won't be touched by non-docker clusters. I'm a little bit confused as we does not support docker with TPU for now?

Yes, testing with docker as runtime. Right, docker on TPU was not supported, but IIRC, it was mainly blocked by the exact issue you fixed, see here: https://github.com/skypilot-org/skypilot/pull/3819/files

Maybe we can pick that PR up once this is merged.

**Comment by cblmemo** [comment]
@Michaelvll Just tested all and it works well. On Lambda it failed to recognize the nvidia runtime for some reason, so i manually patched the run options. PTAL again!

https://github.com/skypilot-org/skypilot/blob/71a95f4bf7f1446e80bb5c24d23c1695bc4fc031/sky/provision/docker_utils.py#L375-L393

File: sky/utils/command_runner.py
Comment (Michaelvll) [inline sky/utils/command_runner.py:506]: Can we comment here for what output are we eliminating?

**Review by Michaelvll — APPROVED** [review]
Thanks for the update and adding TPU docker support @cblmemo! LGTM.

**Pull request merged**
